src/Compilers/VisualBasic/Portable/Compilation/SpeculativeSemanticModelWithMemberModel.vb

```diff
@@ -0,0 +1,305 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports System.Runtime.InteropServices
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic
+
+    ''' <summary>
+    ''' Instances of this type represent user-facing speculative semantic models that are backed by 
+    ''' internal <see cref="MemberSemanticModel"/>.
+    ''' </summary>
+    Friend NotInheritable Class SpeculativeSemanticModelWithMemberModel
+        Inherits PublicSemanticModel
+
+        Private ReadOnly _parentSemanticModel As SyntaxTreeSemanticModel
+        Private ReadOnly _speculatedPosition As Integer
+        Private ReadOnly _memberModel As MemberSemanticModel
+
+        Private Sub New(parentSemanticModel As SyntaxTreeSemanticModel,
+                        speculatedPosition As Integer)
+            Debug.Assert(Not parentSemanticModel.IsSpeculativeSemanticModel, VBResources.ChainingSpeculativeModelIsNotSupported)
```

Comment:
Nit: I don't think we should use a localized resource for a debug.assert. If it fails on the spanish test leg, we'll get a spanish assert message.

Reply:
> Nit: I don't think we should use a localized resource for a debug.assert. If it fails on the spanish test leg, we'll get a spanish assert message.

This is an adjusted copy of an original assert (definitely impossible conditions were removed). 
```Debug.Assert(parentSemanticModelOpt Is Nothing OrElse Not parentSemanticModelOpt.IsSpeculativeSemanticModel, VBResources.ChainingSpeculativeModelIsNotSupported)```

I'll leave the code as is, it is unlikely the assert will fail only on spanish leg, and, even if that happens, there will be a call stack clearly pointing at the root cause. Most of our asserts do not include any clarifying messages.